cpp/src/copying/utilities/copying_utils.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2019 BlazingDB, Inc.
+ *     Copyright 2019 Christian Noboa Mardini <christian@blazingdb.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "copying/utilities/copying_utils.hpp"
+#include "cudf.h"
+#include "rmm/rmm.h"
+#include "utilities/error_utils.hpp"
+
+namespace cudf {
+namespace utilities {
+
+CudaVariableScope::CudaVariableScope(cudaStream_t stream)
+: counter_{nullptr}, stream_{stream} {
+  RMM_ALLOC((void**)&counter_, sizeof(gdf_size_type), stream_);
+  CUDA_TRY(cudaMemsetAsync(counter_, 0, sizeof(gdf_size_type), stream_));
+}
+
+CudaVariableScope::~CudaVariableScope() {
+  RMM_FREE(counter_, stream_);
+}
+
+gdf_size_type* CudaVariableScope::get_pointer() {
+  return counter_;
+}
+
+void CudaVariableScope::load_value(gdf_size_type& value) {
+  CUDA_TRY(cudaMemcpyAsync(&value, counter_, sizeof(gdf_size_type), cudaMemcpyDeviceToHost, stream_));
+}
+
+
+BaseCopying::BaseCopying(gdf_column const*   input_column,
+                         gdf_column const*   indexes,
+                         cudf::column_array* output_columns,
+                         cudaStream_t*       streams,
+                         gdf_size_type       streams_size)
+: input_column_{input_column},
+  indexes_{indexes},
+  output_columns_{output_columns},
+  streams_{streams},
+  streams_size_{streams_size}
+{ }
+
+BaseCopying::KernelOccupancy BaseCopying::calculate_kernel_data_occupancy(gdf_size_type size) {
```

Comment:
All this looks redundant with https://github.com/rapidsai/cudf/blob/branch-0.7/cpp/src/utilities/cuda_utils.hpp#L39

Reply:
I used the suggested function